File: frontend/src/api/agent.ts
import request from '@/utils/request.ts'
import type { ICommonResponse } from '@/types/common.ts'

export interface AgentItem {
  id: number
  name: string
  contact: string
  parentId: number
  level: number
  discount: number
  balance: number
  status: number
  createdAt: string
  updatedAt: string
  children?: AgentItem[]
}

export const getAgentList = () => {
  return request.get<ICommonResponse<{ agents: AgentItem[] }>>('/admin/agent/list')
}

export const getAgent = (id: number) => {
  return request.get<ICommonResponse<{ agent: AgentItem }>>(`/admin/agent/${id}`)
}

export const createAgent = (data: { name: string; contact: string; parentId: number; discount: number }) => {
  return request.post<ICommonResponse<{ id: number }>>('/admin/agent/create', data)
}

export const updateAgent = (data: { id: number; name: string; contact: string; discount: number; status: number }) => {
  return request.post<ICommonResponse<null>>('/admin/agent/update', data)
}

export const deleteAgent = (id: number) => {
  return request.post<ICommonResponse<null>>('/admin/agent/delete', { id })
}

File: backend/internal/dao/agent.go
package dao

import "time"

type Agent struct {
	ID        uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	Name      string    `gorm:"type:varchar(100);not null" json:"name"`
	Contact   string    `gorm:"type:varchar(200)" json:"contact"`
	ParentID  uint      `gorm:"not null;default:0;index" json:"parentId"`
	Level     uint8     `gorm:"not null;default:0" json:"level"` // 0=总代 1=一级 2=二级 3=三级
	Discount  float64   `gorm:"type:decimal(5,2);not null;default:100.00" json:"discount"`
	Balance   float64   `gorm:"type:decimal(12,2);not null;default:0" json:"balance"`
	Status    uint8     `gorm:"not null;default:0" json:"status"` // 0=正常 1=冻结 2=禁用
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

func (Agent) TableName() string { return "agents" }
